pkg/sources/gcs/gcs.go

```diff
@@ -147,142 +152,6 @@ func setGCSManagerOptions(include, exclude []string, includeFn, excludeFn func([
-
-func (p *progressInfo) setProcessStatus(obj object, fn progressStateFn) {
-	fn(obj.bucket, obj.name, p)
-}
-
-func setProcessingBucketObject(bucket, obj string, progress *progressInfo) {
-	progress.processing(bucket, obj)
-}
-
-func (p *progressInfo) processing(bkt, obj string) {
-	p.bucketObjects[bkt].Processing[obj] = struct{}{}
-}
-
-func setProcessedBucketObject(bucket, obj string, progress *progressInfo) {
-	progress.processed(bucket, obj)
-}
-
-func (p *progressInfo) processed(bkt, obj string) {
-	// Remove the object from the processing list and increment the processed count.
-	delete(p.bucketObjects[bkt].Processing, obj)
-
-	p.bucketObjects[bkt].ProcessedCnt++
-	// If all objects in the bucket have been processed, mark the bucket as processed.
-	if p.bucketObjects[bkt].ProcessedCnt == p.bucketObjects[bkt].TotalCnt {
-		p.bucketObjects[bkt].IsBucketProcessed = true
-		p.processedBucketCnt++
-	}
-	p.processedObjectsCnt++
-
-	// If the object is not greater (lexicographically) than the last processed object, we can skip it.
-	// This ensures we keep the greatest object name as the last processed object.
-	if obj < p.bucketObjects[bkt].Processed {
-		return
-	}
-	p.bucketObjects[bkt].Processed = obj
-}
-
-func setResumeBucketOffset(s string) (map[string]offsetInfo, error) {
-	resumeInfo := progressInfo{}
-	if s != "" {
-		if err := json.Unmarshal([]byte(s), &resumeInfo.bucketObjects); err != nil {
-			return nil, fmt.Errorf("error unmarshalling resume info: %w", err)
-		}
-	}
-
-	return calcBktOffset(resumeInfo.bucketObjects)
-}
-
-// In order to calculate the bucket offset, we need to know the last object
-// that was processed for each bucket, as well as any objects that were processing.
-// If the bucket was fully processed, we don't need to set an offset.
-// If there were no objects processing, we can just use the last object that was processed.
-// If there were objects processing, we need to find the first object (lexicographically)
-// that was processing, as that is the next object that needs to be processed.
-func calcBktOffset(resumeInfo map[string]*objectsProgress) (map[string]offsetInfo, error) {
-	bucketOffset := make(map[string]offsetInfo, len(resumeInfo))
-	for objName, progress := range resumeInfo {
-		info := offsetInfo{}
-		if progress.IsBucketProcessed {
-			info.isBucketProcessed = true
-			bucketOffset[objName] = info
-			continue
-		}
-		if len(progress.Processing) == 0 {
-			info.lastProcessedObject = progress.Processed
-			bucketOffset[objName] = info
-			continue
-		}
-
-		processing := make([]string, 0, len(progress.Processing))
-		for k := range progress.Processing {
-			processing = append(processing, k)
-		}
-		sort.Strings(processing)
-		info.lastProcessedObject = processing[0]
-		bucketOffset[objName] = info
-	}
-
-	return bucketOffset, nil
-}
-
 // enumerate all the objects and buckets in the source and use the results to
 // set the progress information. This will be used track progression of the scan,
```

Comment:
There's no more progress information, right? Does that make this comment out of date now?

Reply:
ah, yep. Updated it to match the new behavior in this PR. Thanks.